Mage.Sets/src/mage/cards/s/ShareTheSpoils.java

```diff
@@ -0,0 +1,348 @@
+package mage.cards.s;
+
+import mage.MageIdentifier;
+import mage.MageObjectReference;
+import mage.abilities.Ability;
+import mage.abilities.TriggeredAbility;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.effects.*;
+import mage.cards.*;
+import mage.constants.*;
+import mage.game.CardState;
+import mage.game.ExileZone;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.players.ManaPoolItem;
+import mage.players.Player;
+import mage.players.PlayerList;
+import mage.util.CardUtil;
+import mage.watchers.Watcher;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * @author Alex-Vasile
+ */
+public final class ShareTheSpoils extends CardImpl {
+
+    // Number used to ensure that each cast of Share the Spoils has its own unique exile pile
+    //      "Once Share the Spoils leaves the battlefield, players may no longer play the exiled cards.
+    //       If Share the Spoils is destroyed and somehow returns to the battlefield,
+    //       it’s a new object with no memory of the previously exiled cards."
+    private final String exileZoneIdString;
```

Comment:
If you want an `exileId` that's unique to the permanent then use `CardUtil.getExileZoneId(source, game)`, it uses the source object's UUID and zone change counter. Also you shouldn't store non-static variables on a card class as it would cause problems if you tried to e.g. clone the permanent.